iroha/src/crypto.rs

```diff
@@ -37,11 +33,20 @@ pub struct Signature {
 }
 
 impl Signature {
-    pub fn new(public_key: PublicKey, signature: Ed25519Signature) -> Signature {
-        Signature {
+    pub fn new(
+        public_key: PublicKey,
+        payload: &[u8],
+        private_key: &PrivateKey,
+    ) -> Result<Signature, String> {
+        let transaction_signature = Signer::new(&Ed25519Sha512, &private_key)
```

Comment:
Ed25519Sha512 is fine for our MVP, but long term, we should allow people to set any crypto supported by URSA

Reply:
https://soramitsu.atlassian.net/browse/RH2-71